test/compiler/irpasses.jl

```diff
@@ -342,3 +342,15 @@ let code = code_typed(pi_on_argument, Tuple{Any})[1].first.code,
     @test nisa == 1
     @test found_pi
 end
+
+# Check that pure functions with non-inlineable results still get deleted
+struct Big
+    x::NTuple{1024, Int}
+end
+@Base.pure Big() = Big(ntuple(identity, 1024))
+function pure_elim_full()
+    Big()
+    nothing
+end
+
+@test length(code_typed(pure_elim_full, Tuple{})[1][1].code) == 1
```

Comment:
This seems awfully brittle (partly because I know these are always failing on the coverage tests). We're you improving this in https://github.com/JuliaLang/julia/pull/38071/files#diff-1ebb351fd12a98d8b417ff4e81ccdd32a9a240dfcfe583f3a25c71a7efd224ae, so it should perhaps go in that file?